packages/composites/anchor/src/utils.ts

```diff
@@ -1,39 +1,67 @@
 import { AdapterContext, AdapterRequest, AdapterResponse } from '@chainlink/types'
-import * as TA from '@chainlink/token-allocation-adapter'
-import { BigNumber } from 'ethers'
-import { DEFAULT_TOKEN_DECIMALS } from './config'
+import { BigNumber, ethers } from 'ethers'
+import { FIXED_POINT_DECIMALS } from './config'
+import * as view from '@chainlink/terra-view-function-adapter'
+import { AdapterError } from '@chainlink/ea-bootstrap'
 
 export const getTokenPrice = async (
   input: AdapterRequest,
   context: AdapterContext,
-  symbol: string,
-  decimals = DEFAULT_TOKEN_DECIMALS,
-): Promise<AdapterResponse> => {
-  const _config = TA.makeConfig()
-  const _execute = TA.makeExecute(_config)
-  const allocations = [
-    {
-      symbol,
-      balance: BigNumber.from(10).pow(decimals).toString(),
-      decimals,
-    },
-  ]
-  return await _execute({ id: input.id, data: { ...input.data, allocations } }, context)
+  feedAddress: string,
+  feedDecimals = 8,
+): Promise<ethers.BigNumber> => {
+  const feedResponse = await callViewFunctionEA(input, context, feedAddress, {
+    aggregator_query: { get_latest_round_data: {} },
+  })
+  const latestAnswer = feedResponse.data.result.answer
+  const result = ethers.utils.parseUnits(latestAnswer, feedDecimals)
+  throwErrorForInvalidResult(input.id, result, `Chainlink Terra feed ${feedAddress}`)
+  return result
 }
 
 export const convertUSDQuote = async (
   input: AdapterRequest,
   context: AdapterContext,
-  usdPrice: number,
-  targetQuote: string,
-  targetQuoteDecimals = DEFAULT_TOKEN_DECIMALS,
-): Promise<number> => {
-  const targetQuoteUSDRateResp = await getTokenPrice(
+  usdPrice: ethers.BigNumber,
+  targetQuoteFeedAddress: string,
+  feedDecimals = 8,
+): Promise<ethers.BigNumber> => {
+  const targetQuoteUSDRate = await getTokenPrice(
     input,
     context,
-    targetQuote,
-    targetQuoteDecimals,
+    targetQuoteFeedAddress,
+    feedDecimals,
   )
-  const targetQuoteUSDRate = targetQuoteUSDRateResp.data.result
-  return usdPrice / targetQuoteUSDRate
+  return usdPrice.mul(BigNumber.from(10).pow(FIXED_POINT_DECIMALS)).div(targetQuoteUSDRate)
+}
+
+export const callViewFunctionEA = async (
+  input: AdapterRequest,
+  context: AdapterContext,
+  address: string,
+  query: unknown,
+): Promise<AdapterResponse> => {
+  const _config = view.makeConfig()
+  const _execute = view.makeExecute(_config)
+  const viewFunctionAdapterRequest: AdapterRequest = {
+    id: input.id,
+    data: {
+      address,
+      query,
+    },
+  }
+  return await _execute(viewFunctionAdapterRequest, context)
+}
+
+export const throwErrorForInvalidResult = (
+  jobRunID: string,
+  value: ethers.BigNumber,
+  label: string,
+): void => {
+  if (value.eq(BigNumber.from('0')))
```

Comment:
Might be better to check <= 0?

```suggestion
  if (value.lte(0))
```